x-pack/plugins/ingest_manager/server/services/epm/registry/index.ts

```diff
@@ -203,16 +206,21 @@ async function getOrFetchArchiveBuffer(pkgName: string, pkgVersion: string): Pro
 }
 
 export async function ensureCachedArchiveInfo(name: string, version: string) {
-  const pkgkey = getCacheKey(`${name}-${version}`);
-  if (!cacheHas(pkgkey)) {
+  const pkgkey = getArchiveKey(name, version);
+  if (!pkgkey || !cacheHas(pkgkey)) {
     await getArchiveInfo(name, version);
   }
 }
 
 async function fetchArchiveBuffer(pkgName: string, pkgVersion: string): Promise<Buffer> {
   const { download: archivePath } = await fetchInfo(pkgName, pkgVersion);
-  const registryUrl = getRegistryUrl();
-  return getResponseStream(`${registryUrl}${archivePath}`).then(streamToBuffer);
+  const archiveUrl = `${getRegistryUrl()}${archivePath}`;
+  const buffer = await getResponseStream(archiveUrl).then(streamToBuffer);
+
+  setArchiveKey(pkgName, pkgVersion, archivePath);
+  cacheSet(archivePath, buffer);
```

Comment:
ensure we always cache a fetched archive buffer

Reply:
is this cache invalidated if the current package is removed? This is mainly important for package development.

Reply:
@ruflin No, but that's the same behavior as now. I'm not sure if this means it is an issue or if it's being avoided/worked around in some other way.

Reply:
@skh @neptunian I thought one of you solve this issue recently or at least debugged it?

Reply:
There is https://github.com/elastic/kibana/issues/68890 -- it's still open because I misunderstood the original description and tested the wrong things, and then it got moved back in the queue.

Reply:
@skh Thanks for the link. This basically means we can move forward with what we have in this PR and will follow up later in #68890